src/Maestro/Maestro.Web/Api/v2019_01_16/Controllers/BuildTimeController.cs

```diff
@@ -0,0 +1,128 @@
+        /// This is captured for generating the longest build times for the dependency flow graph
+        /// </summary>
+        /// <param name="id">Default channel id</param>
+        [HttpGet("{id}")]
+        [SwaggerApiResponse(HttpStatusCode.OK, Type = typeof(BuildTime), Description = "Gets the average official build time and average pr build time for a given default channel")]
+        [ValidateModelState]
+        public virtual async Task<IActionResult> GetBuildTimes([Required]int id)
+        {
+            Data.Models.DefaultChannel defaultChannel = await _context.DefaultChannels.FindAsync(id);
+
+            if (defaultChannel == null)
+            {
+                return NotFound();
+            }
+
+            var properties = new ClientRequestProperties();
+
+            // There is a mismatch between how we store the names of the repositories in BAR
+            // and how we store them in Kusto. Normalize to the last part of the url.
+            properties.SetParameter("_Repository", defaultChannel.Repository.Split("/").Last());
+            properties.SetParameter("_SourceBranch", defaultChannel.Branch);
+
+            string queryParameters = String.Join(",", properties.Parameters.Select(p => $"{p.Key}:{KustoDataTypes.String}"));
+
+            // We only care about builds that complete successfully or partially successfully 
+            // from the given repository. We summarize duration of the builds over the last 7 days.
+            // There are multiple different definitions that run in parallel, so we summarize
+            // on the definition id and ultimately choose the definition that took the longest.
+            string commonQueryText = @"| where Repository endswith _Repository
+                | where Result != 'failed' and Result != 'canceled' 
+                | where FinishTime > ago(7d) 
+                | extend duration = FinishTime - StartTime 
+                | summarize avg(duration) by DefinitionId
+                | summarize max(avg_duration)";
+
+            // We only want the pull request time from the public ci. We don't exclude on branch,
+            // as all PRs come in as refs/heads/#/merge rather than what branch they are trying to
+            // apply to.
+            string publicQuery = $@"TimelineBuilds 
+                | where Project == 'public'
+                | where Reason == 'pullRequest' 
+                {commonQueryText}";
+
+            // For the official build times, we want the builds that were generated as a CI run 
+            // (either batchedCI or individualCI) for a specific branch--i.e. we want the builds
+            // that are part of generating the product.
+            string internalQuery = $@"TimelineBuilds 
+                | where Project == 'internal' 
+                | where Reason contains 'CI'
+                | where SourceBranch endswith _SourceBranch
```

Comment:
Will this mix up builds of "internal/release/3.1.1xx" and "release/3.1.1xx" depending on the default channel?

Reply:
Yes, that would confuse the two. I could change this to SourceBranch == 'ref/heads/_SourceBranch', which would cover that.

Reply:
Yeah I'd say just strip away the refs/heads section of branch names (there is an API we use to do this in darclib already) and then compare exact strings.

Reply:
There's an api that would strip it away in the Kusto query? The problem is that in Kusto, we store refs/heads/.

Reply:
Ahh no...i mean on the client side.

Reply:
Gotcha. I worry that this could break the Power BI reports because they had to do some connections between Kusto and Maestro, and I assume there was some munging of the data. I'd want to check with @chcosta before changing how we store things in Kusto.

Reply:
Okay, let's add an issue in arcade to deal with this.

Reply:
Sounds good. I also remembered while talking with Epsi and Megan than github throttles super hard when you try to programmatically ping their api too frequently (at least it has in the past for me).

Reply:
Sounds good.

Reply:
Opened https://github.com/dotnet/arcade/issues/4553 to track normalizing the branches in kusto.

Reply:
I decided to just do this in this PR, since I realized that there could be some weirdness if there's a problem just adding refs/heads to the front of the branch name.